app/views/building_map/index.html.erb

```diff
@@ -1,11 +1,141 @@
+	<div class="accordion-item">
+		<h2 class="accordion-header" id="headingOne">
+			<button class="accordion-button collapsed" type="button" data-bs-toggle="collapse"
+				data-bs-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
+				<i class="fas fa-directions directions-icon"></i>
+			</button>
+		</h2>
+		<div id="collapseOne" class="accordion-collapse collapse" aria-labelledby="headingOne"
+			data-bs-parent="#routingInput">
+			<div class="accordion-body">
+				<form class="row gy-2 gx-3 align-items-center" autocomplete="off" id="navigation_form">
+					<div class="col-auto">
+						<div class="input-group">
+							<div class="input-group-text"><i class="fas fa-map-marker-alt route-start-icon"></i></div>
+							<input name='start' type="text" class="form-control" list="startOptions" id="start_input"
+								placeholder="From" onchange="request_location()">
+						</div>
+					</div>
+					<div class="col-auto">
+						<div class="input-group">
+							<div class="input-group-text"><i class="fas fa-map-marker-alt route-end-icon"></i></div>
+							<input name='dest' type="text" class="form-control" list="destOptions" id="dest_input"
+								placeholder="To">
+						</div>
+					</div>
+					<div class="col-auto">
+						<button type="submit" formaction="/building_map" class="btn btn-primary" onclick="request_location()">Go</button>
+					</div>
+				</form>
+			</div>
+		</div>
+	</div>
+</div>
+
+<datalist id="startOptions">
+	<option>Your location</option>
+	<%= Places::DESTINATIONS.each do |dest, coords| %>
+		<option>
+			<%= dest %>
+		</option>
+		<% end %>
+</datalist>
+<datalist id="destOptions">
+	<%= Places::DESTINATIONS.each do |dest, coords| %>
+		<option>
+			<%= dest %>
+		</option>
+		<% end %>
+</datalist>
+
+<%= map(
     center: {
-        # HPI coordinates if no center is set
-        latlng: !@target.nil? ? @target.split(",") : [52.39339, 13.13208],
-        zoom: 18,
+        # HPI coordinates
+        latlng: RoutingHelper.center(@start),
+        zoom: 17,
     },
-    polygons: BuildingMapHelper.leaflet_polygons(),
+    polygons: BuildingMapHelper.leaflet_polygons,
     polylines: BuildingMapHelper.leaflet_polylines(@route),
     markers: BuildingMapHelper.leaflet_markers(@route, @target))
 %>
+
+<script>
+    let currentLocation;
+
+	function validate_start_input() {
+		const start_input = document.getElementById("start_input");
+		const startOptions = document.getElementById("startOptions").options;
+		let valid_start = false;
+		for (option of startOptions) {	
+			if (option.value === start_input.value) valid_start = true;
+		}
+		return valid_start;	
+	}
+
+	function validate_dest_input() {
+		const dest_input = document.getElementById("dest_input");
+		const destOptions = document.getElementById("destOptions").options;
+		let valid_dest = false;
+		for (option of destOptions) {	
+			if (option.value === dest_input.value) valid_dest = true;
+		}
+		return valid_dest;
+	}
```

Comment:
The duplication could be removed by  having a `function validate_place_input(inputId, optionsId) { ... }` or even one where you pass the DOM elements returned by `document.getElementById` directly instead of looking them up again (you already do so below)